### PR TITLE
[CALCITE-5952] SemiJoinJoinTransposeRule should check if JoinType supports pushing predicates into its inputs

### PR DESCRIPTION
**Pull request by LeonidChistov**
https://issues.apache.org/jira/browse/CALCITE-5952

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rules/SemiJoinJoinTransposeRule.java [hunk 1]
```diff
@@ -114,6 +114,18 @@ public SemiJoinJoinTransposeRule(RelBuilderFactory relBuilderFactory) {
       return;
     }
 
+    // join type need to allow pushing predicate (represented as semi-join in our case)
```
Comment (rubenada) [inline core/src/main/java/org/apache/calcite/rel/rules/SemiJoinJoinTransposeRule.java:117]: minor: need**s**

**Comment by rubenada** [comment]
Thanks for your reactivity @LeonidChistov !
LGTM
I'll merge in 24-48h if no other remark appears.

**Comment by rubenada** [comment]
@LeonidChistov sorry to bother you, it seems due to a recent force push on Calcite main, the PR now shows many unrelated commits; could you please rebase to latest main?

**Comment by LeonidChistov** [comment]
@rubenada Done.

**Pull request merged**